check/pytest

```diff
@@ -19,15 +19,21 @@ cd "$(git rev-parse --show-toplevel)"
 
 PYTEST_ARGS=()
 ACTUALLY_QUIET=""
+RESOURCE_ESTIMATES=""
 for arg in $@; do
     if [[ "${arg}" == "--actually-quiet" ]]; then
         ACTUALLY_QUIET=1
+    elif [[ "${arg}" == "--resource-estimates" ]]; then
+        RESOURCE_ESTIMATES=1
     else
         PYTEST_ARGS+=("${arg}")
     fi
 done
 
-if [ -z "${ACTUALLY_QUIET}" ]; then
+
+if [ "${RESOURCE_ESTIMATES}" -eq 1 ]; then
+    pytest -c dev_tools/conf/pytest.ini -m "not slow" "${PYTEST_ARGS[@]}" src/openfermion/resource_estimates
```

Comment:
Doesn't pytest just go through all the source files and run anything that looks like a test? If I run pytest *without* the `--resource-estimates` flag ... won't it run the resource estimation tests anyways.

Please provide an alternative rationale if you have one: but I would think that we would always tell pytest to try to test all the test files it can and your mechanism of skipping tests if the dependencies aren't installed will just cause those to be skipped if the dependencies aren't installed. That is -- instead of relying on this flag we just rely on the precense of the dependencies.

If you *do* want a flag: the behavior I would expect is a sort of tri-state logic: If resource estimates are disabled, never run the tests regardless of presence of dependencies. If it's "auto" (the default) run the tests if the dependencies exist, otherwise they will be skipped with your skipping mechanism. If it's set to "on": always try to run the tests and treat the skips as failures.

This probably isn't worth the effort, so I'd just remove the flag and always use the "auto" behavior I've described

Reply:
good point. I was having some difficulty passing literal quotes through to check/pytest (-m "not slow")

Reply:
Fixed this "$@" vs $@